### PR TITLE
movement.is_moving works, add event 'movement_follow_path_done'

### PR DESCRIPTION
**Pull request by luken**

### CODE REVIEW

File: spockbot/plugins/helpers/movement.py [hunk 1]
```diff
@@ -19,7 +23,10 @@ def stop(self):
 
     @property
     def is_moving(self):
-        return self.__plug.path_nodes is not None
+        is_processing = self.__plug.pathfinding.is_processing()
+        is_moving = self.__plug.path_nodes is not None
+        ret = is_moving or is_processing
+        return ret
```
Comment (Gjum) [inline spockbot/plugins/helpers/movement.py]: probably want to inline `ret`

File: spockbot/plugins/helpers/movement.py [hunk 2]
```diff
@@ -92,4 +99,6 @@ def follow_path(self, _, __):
         if self.physics.move_target(target) or jumped:
             self.path_nodes.popleft()
             if not self.path_nodes:
+                self.event.emit('movement_follow_path_done')
```
Comment (Gjum) [inline spockbot/plugins/helpers/movement.py]: Emitting this before stopping prevents a handler to start another movement.

File: spockbot/plugins/helpers/pathfinding.py [hunk 3]
```diff
@@ -77,6 +81,9 @@ def build_list_from_node(self, node):
             ret.appendleft(ret[0].parent)
         return ret
 
+    def is_processing(self):
```
Comment (Gjum) [inline spockbot/plugins/helpers/pathfinding.py]: `@property`

File: spockbot/plugins/helpers/movement.py (hunk 2 above)
Comment (luken) [inline spockbot/plugins/helpers/movement.py]: good catch, thanks.

**Comment by luken** [comment]
Getting old, will resubmit after reworking.

**Pull request closed without merging**